src/crepe/manager/LoopTimerManager.h

```diff
@@ -0,0 +1,173 @@
+#pragma once
+
+#include <chrono>
+
+#include "Manager.h"
+
+namespace crepe {
+
+/**
+ * \brief Manages timing and frame rate for the game loop.
+ * 
+ * The LoopTimerManager class is responsible for calculating and managing timing functions 
+ * such as delta time, frames per second (FPS), fixed time steps, and time scaling. It ensures 
+ * consistent frame updates and supports game loop operations, such as handling fixed updates 
+ * for physics and other time-sensitive operations.
+ */
+class LoopTimerManager : public Manager {
+public:
+	/**
+	 * \param mediator A reference to a Mediator object used for transfering managers.
+	 */
+	LoopTimerManager(Mediator & mediator);
+	/**
+	 * \brief Get the current delta time for the current frame.
+	 *	
+	 * This value represents the estimated frame duration of the current frame.
+	 * This value can be used in the frame_update to convert pixel based values to time based values.
+	 * 
+	 * \return Delta time in seconds since the last frame.
+	 */
+	double get_delta_time() const;
+
+	/**
+	 * \brief Get the current game time.
+	 *
+	 * \note The current game time may vary from real-world elapsed time. It is the cumulative
+	 * sum of each frame's delta time.
+	 *
+	 * \return Elapsed game time in seconds.
+	 */
+	double get_current_time() const;
+
+	/**
+	 * \brief Set the target frames per second (FPS).
+	 *
+	 * \param fps The desired frames rendered per second.
+	 */
+	void set_target_fps(int fps);
+
+	/**
+	 * \brief Get the current frames per second (FPS).
+	 *
+	 * \return Current FPS.
+	 */
+	int get_fps() const;
+
+	/**
+	 * \brief Get the current time scale.
+	 *
+	 * \return The current time scale, where (0 = pause, < 1 = slow down, 1 = normal speed, > 1 = speed up).
+	 * up the game.
+	 */
+	double get_time_scale() const;
+
+	/**
+	 * \brief Set the time scale.
+	 *
+	 * time_scale is a value that changes the delta time that can be retrieved using get_delta_time function. 
+	 * 
+	 * \param time_scale The desired time scale (0 = pause, < 1 = slow down, 1 = normal speed, > 1 = speed up).
+	 */
+	void set_time_scale(double time_scale);
+
+	/**
+	 * \brief Get the fixed delta time in seconds without scaling by the time scale.
+	 *
+	 * This value is used in the LoopManager to determine how many times 
+	 * the fixed_update should be called within a given interval.
+	 * This value is also the timing value which is used in the fixed_loop to convert pixels to time.
+	 *
+	 * \return The unscaled fixed delta time in seconds.
+	 */
+	double get_fixed_delta_time() const;
+
+	/**
+	 * \brief Set the fixed_delta_time in seconds.
+	 * 
+	 * \param seconds fixed_delta_time in seconds.
+	 * 
+	 * The fixed_delta_time value is used to determine how many times per second the fixed_update and process_input functions are called.
+	 * This value is also the timing value which is used in the fixed_loop to convert pixels to time.
+	 */
+	void set_fixed_delta_time(double seconds);
```

Comment:
Waarom kun je fixed delta time instellen? Check ook nog even alle doxygen comments want die zeggen nu dat de frequentie waarmee de fixed update uitgevoerd wordt afhankelijk is van deze waarde, wat niet klopt. Er staat ook dat je pixels omrekent naar tijd?